### PR TITLE
New utility: Enmax

### LINKED ISSUE #17: New provider: Enmax (Canada)
Good afternoon, I'm super eager to use this library (as part of home-assistant) but my python abilities are lacking. If I can provide all of the information from a browser session to login is there someone that can help add/validate the new provider? I'd be happy to assist in any-way that I can. 

the subdomain that I see the requests going through are at enmx.opower.com (the missing a is intentional) and the authenticate urls are also mentioning the sso at sso.enmax.com.

When I log into my provider's website at enmax.com I am never presented with a url in my browser visible at opower.com but i see many hits to the enmx.opower.com api in the network tab of chrome's dev tools.

### PR DESCRIPTION
**Pull request by aman207**
Closes #17

### CODE REVIEW

File: src/opower/utilities/enmax.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                raise InvalidAuth(result['ErrorMessage'])
+
+        #Get authorization token for opower
+        async with session.post(
+            "https://www.enmax.com/YourAccountSite/_vti_bin/Enmax.Internet.Opower/MyEnergyIQService.svc/IssueAccessToken",
+            headers={"User-Agent": USER_AGENT},
+            raise_for_status=True,
+        ) as resp:
+            token = await resp.text()
+            return str(token).replace('"', '')
```
Comment (tronikos) [inline src/opower/utilities/enmax.py:85]: nit: add new line at the end of file

File: src/opower/utilities/enmax.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
+"""Enmax"""
```
Comment (tronikos) [inline src/opower/utilities/enmax.py:1]: nit: Add an empty line after this and add missing period.

Run:
`isort . ; black . ; flake8 . ; ruff . --fix ; mypy --install-types .`

File: src/opower/utilities/enmax.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+    @staticmethod
+    async def async_login(
+        session: aiohttp.ClientSession,
+        username: str,
+        password: str,
+        optional_mfa_secret: Optional[str],
+    ) -> None:
+
+        #Get request digest (required for authentication to Enmax)
```
Comment (tronikos) [inline src/opower/utilities/enmax.py:39]: Space after #. Same below at line 58 and 78.

File: src/opower/utilities/enmax.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            headers={
+                "User-Agent": USER_AGENT,
+                "Content-Type": 'text/xml',
+                },
+            data=b'<?xml version="1.0" encoding="utf-8"?><soap:Envelope xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">  <soap:Body>    <GetUpdatedFormDigest xmlns="http://schemas.microsoft.com/sharepoint/soap/" />  </soap:Body></soap:Envelope>',
+            raise_for_status=True,
+        ) as resp:
+            xml = await resp.text()
+
+        xml = ET.fromstring(xml)
```
Comment (tronikos) [inline src/opower/utilities/enmax.py:51]: mypy might complain for reusing the same variable name. Use a different one.

File: src/opower/utilities/enmax.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        for i in xml.iter():
+            if(i.tag == "{http://schemas.microsoft.com/sharepoint/soap/}GetUpdatedFormDigestResult"):
```
Comment (tronikos) [inline src/opower/utilities/enmax.py:53]: No parenthesis for conditions unless they are multiline. black will likely fix this for you.

File: src/opower/utilities/enmax.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+                requestdigest = i.text
```
Comment (tronikos) [inline src/opower/utilities/enmax.py:54]: break?

File: src/opower/utilities/enmax.py (hunk 5 above)
Comment (tronikos) [inline src/opower/utilities/enmax.py:53]: What's the point of the curly brackets here?

File: src/opower/utilities/enmax.py (hunk 1 above)
Comment (tronikos) [inline src/opower/utilities/enmax.py:85]: Is casting to str necessary? Isn't it str to begin with?

**Review by tronikos — CHANGES_REQUESTED** [review]

**Comment by tronikos** [comment]
Can you also update the README? Once this is merged, can you create a virtual HA integration?

File: src/opower/utilities/enmax.py (hunk 5 above)
Comment (aman207) [reply src/opower/utilities/enmax.py]: black didn't like the line length so it split it up

Comment (aman207) [reply src/opower/utilities/enmax.py]: this is the exact string that gets returned by `xml.iter()`. If it remove them, it doesn't work

**Comment by aman207** [comment]
> Can you also update the README? Once this is merged, can you create a virtual HA integration?

Yes to both

**Comment by tronikos** [comment]
Thanks!

**Review by tronikos — APPROVED** [review]

**Comment by tronikos** [comment]
I'm bumping the version in https://github.com/home-assistant/core/pull/100501
Please create a virtual integration before the next monthly HA release.

**Comment by aman207** [comment]
PR for that here: https://github.com/home-assistant/core/pull/100503

**Pull request merged**